identity-mdoc/src/main/java/com/android/identity/mdoc/mso/MobileSecurityObjectGenerator.kt

```diff
@@ -260,17 +259,18 @@ class MobileSecurityObjectGenerator(
      * @throws IllegalStateException if required data hasn't been set using the setter
      * methods on this class.
      */
-    fun generate(): ByteArray {
-        check(!digestEmpty) { "Must call addDigestIdsForNamespace before generating" }
-        checkNotNull(mSigned) { "Must call setValidityInfo before generating" }
-        val msoMapBuilder = CborMap.builder()
-        msoMapBuilder.put("version", "1.0")
-        msoMapBuilder.put("digestAlgorithm", mDigestAlgorithm)
-        msoMapBuilder.put("docType", mDocType)
-        msoMapBuilder.put("valueDigests", mValueDigestsOuter.end().build())
-        msoMapBuilder.put("deviceKeyInfo", generateDeviceKeyBuilder().build())
-        msoMapBuilder.put("validityInfo", generateValidityInfoBuilder().build())
-        msoMapBuilder.end()
-        return Cbor.encode(msoMapBuilder.end().build())
-    }
+    fun generate(): ByteArray =
+        CborMap.builder().run {
+            check(!digestEmpty) { "Must call addDigestIdsForNamespace before generating" }
+            checkNotNull(mSigned) { "Must call setValidityInfo before generating" }
+
+            put("version", "1.0")
+            put("digestAlgorithm", mDigestAlgorithm)
+            put("docType", mDocType)
+            put("valueDigests", mValueDigestsOuter.end().build())
+            put("deviceKeyInfo", generateDeviceKeyBuilder().build())
+            put("validityInfo", generateValidityInfoBuilder().build())
+            end()
+            Cbor.encode(end().build())
+        }
```

Comment:
Should only be a single `end()` call here. Happens to work b/c it's an idempotent method but really only should be one!